### ISSUE TITLE
Feature Request: Utilize Python's Thread Module to bake multiple bars at same time

### ISSUE THREAD

**Issue by Brekmister** [gethiox/GXAudioVisualisation#9]
Some people who use blender will have lots of threads and/or cores to work with, why not use it?

Why would this be necessary? Well, because it will speed things up on systems like mine by about 16 times.

Ok, probably nobody will want to implement this but, I am certainly interested in that functionality so, someone may mind as well start learning how threading works :)

**Comment by gethiox** [comment]
@Brekmister indeed, I think it is possible but currently project is not under active development. 
I suggest check [animation nodes](https://www.youtube.com/watch?v=D-y8REUdg6I) plugin which can also handle music visualizer for you, maybe it will be more convenient.

**Comment by Brekmister** [comment]
@gethiox Kinda figured, I started playing with animation nodes and boy, it's super nice. Much faster than GXAV although not multi-threaded. Only downside with animation nodes is that it's much more complicated to make visualizations than GXAV. 

Which is kind of why I will stick with GXAV. I am experimenting with other aspects of blender and I don't want to complicate things more than it has to be.

When I get comfortable with the other stuff Ill switch over and dedicate myself to animation nodes.

Ill leave this open because I can't say it's resolved and it's probably useful for other people to see this. Unless someone bugs me to close it.

**Comment by gethiox** [comment]
Just to inform I plan to rewrite script in the very near future.
I can't be sure if built-in blender's "Bake Sounds to F-Curves" can be used in parallel but i will find out.
Also I really don't like how visualizer looks like with that method, it's too smooth for me.
More decent look can be achieved by [fast fourier transform](https://en.wikipedia.org/wiki/Fast_Fourier_transform) (it is most popular method of frequency visualisation i guess) which i will try to implement (it should be also much faster than blender's built-in feature).
I will try to leave user a choice between this methods of course, but we will see if I will handle that challenge well.
